src/views/dialogs/DetailsDialog/OrderDetailsDialog.tsx

```diff
@@ -39,6 +40,8 @@ export const OrderDetailsDialog = ({ orderId, setIsOpen }: ElementProps) => {
   const selectedLocale = useSelector(getSelectedLocale);
   const isAccountViewOnly = useSelector(calculateIsAccountViewOnly);
   const localCancelOrders = useSelector(getLocalCancelOrders, shallowEqual);
+  const existingOrder = useSelector(getOrderById(orderId), shallowEqual);
```

Comment:
pretty sure this is at least churning references and at worst creating some kind of memory leak

Reply:
I think we should memoize the result of getOrderById so we get one instance per component+orderId.